a1/src/main/java/ro/utcn/spet/a1/repository/jdbc/JdbcQuestionRepository.java

```diff
@@ -0,0 +1,101 @@
+package ro.utcn.spet.a1.repository.jdbc;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+import ro.utcn.spet.a1.model.Question;
+import ro.utcn.spet.a1.model.Tag;
+import ro.utcn.spet.a1.repository.api.QuestionRepository;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@RequiredArgsConstructor
+public class JdbcQuestionRepository implements QuestionRepository {
+
+    private final JdbcTemplate template;
+
+
+    @Override
+    public Question save(Question question) {
+        if(question.getId()!= 0){
+            update(question);
+            updateTags(question);
+        }
+        else{
+            question.setId(insert(question));
+        }
+        return question;
+    }
+
+    @Override
+    public Optional<Question> findById(int id) {
+        List<Question> questions=template.query("SELECT * FROM question WHERE id= ?",
+                new Object[]{ id },
+                new QuestionMapper());
+        return questions.isEmpty() ? Optional.empty() : Optional.of(questions.get(0));
+    }
+
+    @Override
+    public void remove(Question question) {
+        template.update("DELETE FROM question WHERE id= ?",question.getId());
+    }
+
+    @Override
+    public List<Question> findAll() {
+        List<Question> questions=template.query("SELECT * FROM question", new QuestionMapper());
+        for(Question question: questions){
+            question.setTags(findTags(question.getId()));
+        }
+        return questions;
+    }
+
+    @Override
+    public List<Question> findByTitle(String title) {
+        List<Question> questions= template.query("SELECT * FROM question WHERE title LIKE '%?%'",new Object[]{title}, new QuestionMapper());
+        for(Question question: questions){
+            question.setTags(findTags(question.getId()));
+        }
+        return questions;
+    }
+
+    @Override
+    public List<Question> findByTag(Tag tag) {
+        return null;
```

Comment:
Not yet implemented. Still, would be better to return an empty list instead of null :)